sdk/spring/spring-integration-azure-eventhubs/src/main/java/com/azure/spring/integration/eventhubs/inbound/EventHubsInboundChannelAdapter.java

```diff
@@ -108,14 +106,13 @@ public EventHubsInboundChannelAdapter(EventHubsMessageListenerContainer listener
      *
      * @param listenerContainer the event processors container
      * @param listenerMode the listener mode
-     * @param checkpointConfig the checkpoint config
      */
     public EventHubsInboundChannelAdapter(EventHubsMessageListenerContainer listenerContainer,
-                                          ListenerMode listenerMode,
-                                          CheckpointConfig checkpointConfig) {
+                                          ListenerMode listenerMode) {
         this.listenerContainer = listenerContainer;
         this.listenerMode = listenerMode;
-        this.checkpointConfig = checkpointConfig;
+        CheckpointConfig containerCheckpointConfig = listenerContainer.getContainerProperties().getCheckpointConfig();
+        this.checkpointConfig = containerCheckpointConfig == null ? new CheckpointConfig() : containerCheckpointConfig;
```

Comment:
Should we set the default checkpoint mode according to the listener mode?

Reply:
I think we should delegate this check to the checkpoint managers.